src/WorkflowEngine.Sdk/Services/State/WorkflowSemaphoreService.cs

```diff
@@ -201,16 +202,18 @@ async Task ActivateHolderAsync(WorkflowSemaphoreQueueRecord waitingHolder)
             waitingHolder.Raised = true;
             waitingHolder.ExpiresAt = DateTimeOffset.UtcNow.Add(TimeSpan.FromSeconds(waitingHolder.ExpirationAfterSeconds));
             var activatedHolder = await _runtimeTables.WorkflowSemaphoreQueue.UpdateAndReturnAsync(waitingHolder.Id, waitingHolder, cancellationToken);
-            if (activatedHolder.WorkflowInstanceId == myHolder?.WorkflowInstanceId 
+            if (activatedHolder.WorkflowInstanceId == myHolder?.WorkflowInstanceId
                 && activatedHolder.ParentActivityInstanceId == myHolder?.ParentActivityInstanceId
                 && activatedHolder.ParentIteration == myHolder?.ParentIteration)
             {
                 myHolderAsActivated = activatedHolder;
             }
             else
-            {
-                await _requestMgmtCapability.Request.RetryAsync(activatedHolder.WorkflowInstanceId.ToGuidString(),
-                    cancellationToken);
+            {                
+                await WorkflowHelper.RetryAsync(async () =>

```

Comment:
I would like this code to catch the rethrown exception (see my comments in the WorkflowHelper), swallow the exception and log that it happened, so that it is not entirely silent about it.
The log message should probably not be associated with any specific workflow.